# Bioinformatics

## Bioinformatician

-  [What is a bioinformatician](http://blog.fejes.ca/?p=2418)
-  [My Formula as a Bioinformatician](http://computationalproteomic.blogspot.co.uk/2014/01/my-formula-as-bioinformatician.html)
-  [So you want to be a computational biologist?](http://www.nature.com/nbt/journal/v31/n11/abs/nbt.2740.html) ☆
-  [Bioinformatics is not something you are taught, it’s a way of life](http://biomickwatson.wordpress.com/2013/08/06/bioinformatics-is-not-something-you-are-taught-its-a-way-of-life/)
-  [A guide for the lonely bioinformatician](http://biomickwatson.wordpress.com/2013/04/23/a-guide-for-the-lonely-bioinformatician/)
-  [The alternative “what it takes to be a bioinformatician”]( http://biomickwatson.wordpress.com/2013/03/18/the-alternative-what-it-takes-to-be-a-bioinformatician/)
-  [Bioinformatics Curriculum Guidelines: Toward a Definition of Core Competencies](http://www.ploscompbiol.org/article/info:doi%2F10.1371%2Fjournal.pcbi.1003496)
-  [The only core competency you’re ever going to need](http://biomickwatson.wordpress.com/2014/03/10/the-only-core-competency-youre-ever-going-to-need/)
-  [You’re not allowed bioinformatics anymore](http://biomickwatson.wordpress.com/2014/07/21/youre-not-allowed-bioinformatics-anymore/)
-  [Top N Reasons To Do A Ph.D. or Post-Doc in Bioinformatics/Computational Biology](http://caseybergman.wordpress.com/2012/07/31/top-n-reasons-to-do-a-ph-d-or-post-doc-in-bioinformaticscomputational-biology/)
-  [101 Questions: a new series of interviews with notable bioinformaticians](http://www.acgt.me/?tag=101+questions)
-  [生物信息学家级别](http://blog.sciencenet.cn/blog-404304-834869.html) [ Levels of Bioinformatics Research](http://www.longwoodgenomics.org/2014/10/11/levels-of-bioinformatics-research/)
-  [You can code, too!](http://blog.olgabotvinnik.com/blog/2013/11/22/2013-11-22-you-can-code-too/)
-  [It’s time to reboot bioinformatics education](http://toddharris.net/blog/2015/03/23/its-time-to-reboot-bioinformatics-education/)
-  [An Explosion Of Bioinformatics Careers](http://sciencecareers.sciencemag.org/career_magazine/previous_issues/articles/2014_06_13/science.opms.r1400143)

## Bioinformatics

-  [The Phylogeny of Everything, the Origin of Eukaryotes, and the Rules of Taxonomy: Death to Archaea, Bacteria, and Eucarya! Long live Archaebacteria, Eubacteria, Eukaryota, and Prokaryota!](http://judgestarling.tumblr.com/post/79135526488/the-phylogeny-of-everything-the-origin-of-eukaryotes)
-  [Crossroads (iii) – a New Direction for Bioinformatics with Twelve Fundamental Problems](http://www.homolog.us/blogs/blog/2014/10/14/crossroads-iii-a-new-direction-for-bioinformatics-with-twelve-fundamental-problems/)
-  [Ten Simple Rules for Effective Computational Research](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003506)（[高效计算科学研究的十条简单规则](/ten-simple-rules-for-effective-computational-research/)）
-  [Ten Simple Rules for Reproducible Computational Research](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003285)
-  [Ten Simple Rules for the Care and Feeding of Scientific Data](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003542)
-  [An Introduction To Applied Bioinformatics](http://caporasolab.us/An-Introduction-To-Applied-Bioinformatics/)
-  [Freedom in bioinformatics](http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00259/full)
-  [二代测序数据辨（之一）：Clean Data](http://garification.blog.163.com/blog/static/1741330252014624114724457/)
-  [病原微生物高通量测序数据分析笔记](http://indexofire.gitbooks.io/notebook_of_analyzing_pathogen_ngs_data/)
-  [What to do with lots of (sequencing) data](http://ivory.idyll.org/blog//2015-what-to-do-with-sequencing-data.html)
-  [The myths of bioinformatics software](https://liorpachter.wordpress.com/2015/07/10/the-myths-of-bioinformatics-software)

## 生物信息杂谈

-  [《学生物的，不会编程，也可以报考生物信息学的研究生》](http://blog.sciencenet.cn/home.php?mod=space&uid=61772&do=blog&id=590128)by [牛登科](http://blog.sciencenet.cn/home.php?mod=space&uid=61772)。（学生物的，不会编程，也能学生物信息学技术）
-  《[高通量测序能替代PCR吗？](http://blog.sina.com.cn/s/blog_52cb75b90102ekvq.html)》 by [韩建](http://blog.sina.com.cn/jianhan1)
-  [《生物信息学数据分析与皇帝的新装》](http://blog.sciencenet.cn/blog-604918-582915.html)
-  [个性化医疗会带来更昂贵的药物？](http://www.biodiscover.com/news/industry/109583.html)
-  [高通量测序公司靠什么赚钱？](http://blog.sina.com.cn/s/blog_52cb75b90102uw21.html)
-  [生物不退学指南：教你如何靠生物养家糊口](http://mp.weixin.qq.com/s?__biz=MzA3MjQzODAzNA==&mid=200430076&idx=1&sn=35635d8a4b1c13be069ee557f7466983&scene=2&from=timeline&isappinstalled=0#rd) （想进入生物学领域的请看）


## Online courses

-  [Rosalind](http://rosalind.info/) (Rosalind is a platform for learning bioinformatics through problem solving)

## Comprehensive packages

- [Biopython](http://biopython.org/)
- [Biogo](https://github.com/biogo)

## General file formats

- [zindex](https://github.com/mattgodbolt/zindex) - Create an index on a compressed text file
- [tabix](https://github.com/samtools/tabix)  - table file index
- [wormtable](https://github.com/wormtable/wormtable) - Write-once-read-many table for large datasets.

## bam/sam/tabix/bgzf

- [python] [hts-python](https://github.com/brentp/hts-python) - pythonic wrapper for libhts
- [golang] [biogo/hts](https://github.com/biogo/hts)
- [bamtools](https://github.com/pezmaster31/bamtools) - C++ API & command-line toolkit for working with BAM data
- [samblaster](https://github.com/GregoryFaust/samblaster) -  a tool to mark duplicates and extract discordant and split reads from sam files.
- [python] [pysam](ps://github.com/pysam-developers/pysam) - a python module for reading and manipulating Samfiles. It's a lightweight wrapper of the samtools C-API. Pysam also includes an interface for tabix. Another sam parser: [simplesam](https://github.com/mdshw5/simplesam)
- [grabix](https://github.com/arq5x/grabix) - a wee tool for random access into BGZF files
- [golang]  [bix](https://github.com/brentp/bix) - tabix file access with golang using biogo machinery

## Fasta/q

- [seqtk](https://github.com/lh3/seqtk) - Toolkit for processing sequences in FASTA/Q formats
- [python] [pyfaidx](https://github.com/mdshw5/pyfaidx) - pyfaidx: efficient pythonic random access to fasta subsequences
- [golang] [faidx](https://github.com/brentp/faidx)

## GFF/BED/VCF

- [bedtools2](https://github.com/arq5x/bedtools2) - A powerful toolset for genome arithmetic.
- [python] [gffutils](https://github.com/daler/gffutils) - GFF and GTF file manipulation and interconversion
- [python] [pybedtools](https://github.com/daler/pybedtools) - Python wrapper for Aaron Quinlan's BEDTools


- [golang] [irelate](https://github.com/brentp/irelate) - Streaming relation (overlap, distance, KNN) of (any number of) sorted genomic interval sets. #golang
- [golang] [vcfgo](https://github.com/brentp/vcfgo) - a golang library to read, write and manipulate files in the variant call format.

## Others formats

- [blast_table2xml](https://github.com/jameslz/blast_table2xml) - Convert blast m6 format to xml for blast2go

## data structure

- [kvector](https://github.com/olgabot/kvector) - kvector is a small utility for converting motifs to kmer vectors to compare motifs of different lengths

## Models

- [pomegranate](https://github.com/jmschrei/pomegranate) - Graphical models for Python, implemented in Cython for speed.

## Scripts

- [cgat](https://github.com/CGATOxford/cgat) - CGAT - Computational Genomics Analysis Tools
- [bcbb](https://github.com/chapmanb/bcbb) - Incubator for useful bioinformatics code, primarily in Python and R [http://bcbio.wordpress.com](http://bcbio.wordpress.com/)
- [jcvi](https://github.com/tanghaibao/jcvi) - Python utility libraries on genome assembly, annotation and comparative genomics
- [picobio](https://github.com/peterjc/picobio) - Miscellaneous Bioinformatics scripts etc mostly in Python
- [pydna](https://github.com/BjornFJohansson/pydna) - Classes and code for representing double stranded DNA and functions for simulating homologous recombination and Gibson assembly.
- [BioUtils](https://github.com/likit/BioUtils) - Python scripts for miscellaneous bioinformatics stuff.
- [sesbio](https://github.com/sestaton/sesbio) - Bioinformatics scripts for genome analysis
- [ngsutils](https://github.com/ngsutils/ngsutils) - Tools for next-generation sequencing analysis [http://ngsutils.org](http://ngsutils.org/)
- [ngsTools](https://github.com/mfumagalli/ngsTools) - Programs to analyse NGS data for population genetics purposes

## Visualization

- [Circleator](https://github.com/jonathancrabtree/Circleator) - Flexible circular visualization of genome-associated data with BioPerl and SVG.
- [ComplexHeatmap](https://github.com/jokergoo/ComplexHeatmap) - make complex heatmaps as well as self define annotation graphics
- [dalliance](https://github.com/dasmoth/dalliance) - https://github.com/dasmoth/dalliance
- [Question: Which program, tool, or strategy do you use to visualize genomic rearrangements?](https://www.biostars.org/p/158039/) 

### Circos Related

- [Circos](http://circos.ca/): Perl package for circular plots, which are well suited for genomic rearrangements.
- [J-Circos](http://www.australianprostatecentre.org/research/software/jcircos) is a java application for doing interactive work with circos plots.
- [ClicO FS](http://bioinformatics.oxfordjournals.org/content/early/2015/08/18/bioinformatics.btv433.long): an interactive web-based service of Circos.
- [rCircos](https://cran.r-project.org/web/packages/RCircos/index.html): R package for circular plots. [last update: 2013]
-  [OmicCircos](http://www.bioconductor.org/packages/release/bioc/html/OmicCircos.html): R package for circular plots for omics data.[last update: 2015-04]

### Others

- [ggbio](http://bioconductor.org/packages/release/bioc/html/ggbio.html): R package for visualizing biological data. Has a circular view similar to the previous packages.
- [D3 chord diagrams](https://github.com/mbostock/d3/wiki/Chord-Layout) (javascript) can be used to visualize genomic rearrangements. [See this plot of migration flows](http://www.global-migration.info/) as a similar example.
 - [Genomatix Transcriptome Viewer: Gene Fusion analyses](http://www.genomatix.de/online_help/help_eldorado/transcriptome_help.html#gene_fusion_analyses)
- [iFUSE: integrated fusion gene explorer](http://www.ncbi.nlm.nih.gov/pubmed/23661695)
- [FusionAnalyser: a new graphical, event-driven tool for fusion rearrangements discovery](http://www.ncbi.nlm.nih.gov/pubmed/22570408)
- [SOAPFuse includes the option to generate figures](http://soap.genomics.org.cn/soapfuse.html#output2)
- [Gremlin](http://compbio.cs.brown.edu/projects/gremlin/)
- [Seqeyes](http://www.biovis.net/2011/materials/abstracts/BioVispaper107.pdf): A flash tool for visualizing structural variations.
- [SVVIZ](http://svviz.github.io/svviz/) - A READ VISUALIZER TO VALIDATE STRUCTURAL VARIANTS

## Kmer

- [Jellyfish](https://github.com/gmarcais/Jellyfish)


#### Phylogenetic tree

- [R] [ggtree](http://www.bioconductor.org/packages/3.1/bioc/vignettes/ggtree/inst/doc/ggtree.html) - a phylogenetic tree viewer for different types of tree annotations
- [python] [ETE tools](http://etetoolkit.org/)
- [evolview](http://evolgenius.info/evolview/)

## Assembly

- [Bandage](https://github.com/rrwick/Bandage) - a Bioinformatics Application for Navigating De novo Assembly Graphs Easily
- [nucleotid.es](http://nucleotid.es) - an assembler catalogue

## Alignment


- [hpg-aligner](https://github.com/opencb/hpg-aligner) - HPG Aligner is an ultrafast and highly sensitive Next-Generation Sequencing (NGS) mapper which supoprts both DNA and RNA alignment
- [AliView](https://github.com/AliView/AliView) - Software for aligning viewing and editing dna/aminoacid sequences, intuitive, fast and lightweight. Download and website: [http://www.ormbunkar.se/aliview](http://www.ormbunkar.se/aliview)


## Mapping

## Bacterial comparative genomics

- [TOOLS FOR BACTERIAL COMPARATIVE GENOMICS](http://holtlab.net/2015/02/25/tools-for-bacterial-comparative-genomics/)

## Metagenomics

- [shotmap](https://github.com/sharpton/shotmap) - A Shotgun Metagenome Annotation Pipeline
- [metagenomeSeq](https://github.com/HCBravoLab/metagenomeSeq) - Statistical analysis for sparse high-throughput sequencing
- [mmgenome](https://github.com/MadsAlbertsen/mmgenome) - Tools for extracting individual genomes from metagneomes
- [harvest](https://github.com/marbl/harvest) - suite of core-genome alignment and visualization tools for quickly analyzing thousands of intraspecific microbial genomes.
- [PhyloSift](https://github.com/gjospin/PhyloSift) - Phylogenetic and taxonomic analysis for genomes and metagenomes
- [MetaQuery: Annotation and quantitative analysis of genes in the human gut microbiome](http://metaquery.docpollard.org/)

## Chip-seq

- [ChIP-seq-analysis](https://github.com/crazyhottommy/ChIP-seq-analysis)

## Plastform

- [Rabix](https://www.rabix.org/) - Portable Bioinformatics Pipelines
- [bioboxes](https://github.com/bioboxes) - Standards for Interchangeable Bioinformatics Containers
- [Anvi’o](http://merenlab.org/projects/anvio/) is an analysis and visualization platform for ‘omics data. [introduction](http://microbe.net/2015/09/04/anvio-a-new-platform-to-work-with-metagenomic-data/?utm_source=twitterfeed&utm_medium=twitter)

## PCR

- [find_differential_primers](https://github.com/widdowquinn/find_differential_primers) - Scripts to aid the design of differential primers for diagnostic PCR.

## HPC

- [hpcgo](https://github.com/drio/hpcgo) - Helping submit jobs to HPC cluster.
- [easy_qsub](https://github.com/shenwei356/easy_qsub)  - Easily submitting PBS jobs with script template. Multiple input files supported.
